### PR TITLE
Update web3 to 1.2.4 and fix compile errors

### PR DESCRIPTION
**Pull request by justinkook**
Removed unused dependencies from `bitcore-client`.

Bitcore-client was using `scrypt` from `web3`, so manually added `scrypt` and removed `web3`

### CODE REVIEW

File: packages/bitcore-node/src/modules/ethereum/api/csp.ts [hunk 1]
```diff
@@ -13,15 +13,24 @@ import { SpentHeightIndicators } from '../../../types/Coin';
 import { EthListTransactionsStream } from './transform';
 import { ERC20Abi } from '../abi/erc20';
 import { Transaction } from 'web3/eth/types';
-import { EventLog } from 'web3/types';
```
Comment (justinkook) [inline packages/bitcore-node/src/modules/ethereum/api/csp.ts]: `EventLog` type was removed in `web3@1.2.4`

File: packages/bitcore-node/src/modules/ethereum/api/csp.ts [hunk 2]
```diff
@@ -13,15 +13,24 @@ import { SpentHeightIndicators } from '../../../types/Coin';
...
+  blockHash: string;
+  blockNumber: number;
+  raw?: {data: string; topics: any[]};
+}
 interface ERC20Transfer extends EventLog {
   returnValues: {
-    _from: string;
-    _to: string;
-    _value: string;
+    [key: string]: string;
```
Comment (justinkook) [inline packages/bitcore-node/src/modules/ethereum/api/csp.ts]: `EventData[]` returnValues typing is:

```
  [key: string]: any;
```

Not compatible with:
```
  _from: string;
  _to: string;	
  _value: string;
```

File: packages/bitcore-node/src/modules/ethereum/p2p/parityRpc.ts [hunk 3]
```diff
@@ -91,7 +91,7 @@ export class ParityRPC {
 
   public send<T>(data: JsonRPCRequest) {
     return new Promise<T>((resolve, reject) => {
-      this.web3.eth.currentProvider.send(data, function(err, data) {
+      this.web3.eth.currentProvider!['send'](data, function(err, data) {
```
Comment (justinkook) [inline packages/bitcore-node/src/modules/ethereum/p2p/parityRpc.ts]: `this.web3.eth.currentProvider` type is `provider`

```
    | HttpProvider
    | IpcProvider
    | WebsocketProvider
    | string
    | null;
```

The `!` gets rid of `| null`

Each Provider has a `send()` property:
ex. `WebsocketProvider.send(), HttpProvider.send(), ...`

`string` does not have a send property so `['send']`

File: packages/crypto-wallet-core/src/derivation/eth/index.ts [hunk 4]
```diff
@@ -28,7 +28,7 @@ export class EthDeriver implements IDeriver {
     const x = ecPoint.getX().toBuffer({ size: 32 });
     const y = ecPoint.getY().toBuffer({ size: 32 });
     const paddedBuffer = Buffer.concat([x, y]);
-    const address = utils.keccak256(paddedBuffer).slice(26);
+    const address = utils.keccak256(`0x${paddedBuffer.toString('hex')}`).slice(26);
```
Comment (justinkook) [inline packages/crypto-wallet-core/src/derivation/eth/index.ts]: utils.keccak256 param takes in `typeof 'string | BN'`

Cannot pass in type `Buffer`

File: packages/bitcore-node/src/modules/ethereum/api/csp.ts (hunk 2 above)
Comment (justinkook) [reply packages/bitcore-node/src/modules/ethereum/api/csp.ts]: https://github.com/ethereum/web3.js/blob/043cb81d40ede0f6ef1889222c5671f9678e409e/packages/web3-eth-contract/types/index.d.ts#L146

File: packages/bitcore-node/src/modules/ethereum/p2p/parityRpc.ts (hunk 3 above)
Comment (justinkook) [reply packages/bitcore-node/src/modules/ethereum/p2p/parityRpc.ts]: https://github.com/ethereum/web3.js/blob/043cb81d40ede0f6ef1889222c5671f9678e409e/packages/web3-core/types/index.d.ts#L413

File: packages/bitcore-node/src/modules/ethereum/api/csp.ts (hunk 1 above)
Comment (justinkook) [reply packages/bitcore-node/src/modules/ethereum/api/csp.ts]: https://github.com/ethereum/web3.js/blob/043cb81d40ede0f6ef1889222c5671f9678e409e/packages/web3/types/index.d.ts

File: packages/crypto-wallet-core/src/derivation/eth/index.ts (hunk 4 above)
Comment (justinkook) [reply packages/crypto-wallet-core/src/derivation/eth/index.ts]: https://github.com/ethereum/web3.js/blob/a5e832fc20088cf4c4e916bc0ec3107ceb4428d0/packages/web3-utils/types/index.d.ts#L93

File: packages/bitcore-node/src/modules/ethereum/p2p/parityRpc.ts (hunk 3 above)
Comment (micahriggan) [reply packages/bitcore-node/src/modules/ethereum/p2p/parityRpc.ts]: lets do 
```
let provider = this.web3.eth.currentProvider as HttpProvider;
provider.send(data
```

File: packages/bitcore-node/src/modules/ethereum/api/csp.ts [hunk 5]
```diff
@@ -13,15 +13,24 @@ import { SpentHeightIndicators } from '../../../types/Coin';
 import { EthListTransactionsStream } from './transform';
 import { ERC20Abi } from '../abi/erc20';
 import { Transaction } from 'web3/eth/types';
-import { EventLog } from 'web3/types';
 import { partition } from '../../../utils/partition';
 import { WalletAddressStorage } from '../../../models/walletAddress';
-
+import { Contract } from 'web3-eth-contract';
+type ContractAbiInput = ConstructorParameters<typeof Contract>[0];
+interface EventLog {
```
Comment (micahriggan) [inline packages/bitcore-node/src/modules/ethereum/api/csp.ts]: interface EventLog<T> {
   ...,
   returnValues: T;
}

File: packages/bitcore-node/src/modules/ethereum/api/csp.ts (hunk 2 above)
Comment (micahriggan) [reply packages/bitcore-node/src/modules/ethereum/api/csp.ts]:
```
interface ERC20Transfer extends EventLog<{
  _from: string;
  _to: string;	
  _value: string;
}>
```

File: packages/crypto-wallet-core/src/derivation/eth/index.ts (hunk 4 above)
Comment (matiu) [reply packages/crypto-wallet-core/src/derivation/eth/index.ts]: This change seems to be covered but test "should be able to generate a valid ETH address" and "should be able to generate a valid ETH address, privKey, pubKey".  I think we need to include more cases, including edge cases, on the test suite, on a future PR.
